### ISSUE TITLE
Manifest unknown

### ISSUE THREAD

**Issue by Papsnoek** [ThePicklenat0r/hassio-ngrok#23]
Hi, Thanks for the Great Addon. I have been using this to access my Home Assistant and it works Great!

Recently, my addon went missing in HA, and when I removed it to reinstall it, I now get the following error below? Im not sure if its something that has change in HA or just an issue with my installation?

Thanks!

21-01-15 07:53:14 ERROR (SyncWorker_2) [supervisor.docker.interface] Can't install thepicklenat0r/armv7-hassio-ngrok:2.1.5 -> 404 Client Error for http+docker://localhost/v1.41/images/create?tag=2.1.5&fromImage=thepicklenat0r%2Farmv7-hassio-ngrok: Not Found ("manifest for thepicklenat0r/armv7-hassio-ngrok:2.1.5 not found: manifest unknown: manifest unknown").

**Comment by ThePicklenat0r** [comment]
I wasn't able to reproduce that error after uninstalling and reinstalling, the only difference I can see from your error is my machine is amd64 not arm, but I doubt that will make a difference (famous last words LOL)

What version of the OS/Supervisor are you running?

**Comment by bigmat73** [comment]
I have a same error since this afternoon. (arm 64bits)

**Comment by bigmat73** [comment]
Hi, I have a same error since this afternoon. (arm 64bits)
After 01:48PM ngrok addon dont work, impossible to start and this message:
404 Client Error for http+docker://localhost/v1.40/containers/create?name=addon_1ded40ef_ngrok: Not Found ("No such image: thepicklenat0r/aarch64-hassio-ngrok:2.1.5")

Hassio
supervisor 2021.1.4

Thx

**Comment by sheminasalam** [comment]
Hi i am also having the same problem from today morning. Supervisor 2021.1.4

404 Client Error for http+docker://localhost/v1.40/containers/create?name=addon_1ded40ef_ngrok: Not Found ("No such image: thepicklenat0r/aarch64-hassio-ngrok:2.1.5")

**Comment by ThePicklenat0r** [comment]
I was able to reproduce this but not sure what is causing it yet. Still investigating. If anyone has a fix, feel free to submit a PR.

**Comment by ustunsoz** [comment]
I have the same issue:

**Which addon?**
Ngrok Client

**Describe the bug**
Failed to install addon
404 Client Error for http+docker://localhost/v1.40/images/create?tag=2.1.5&fromImage=thepicklenat0r%2Farmv7-hassio-ngrok: Not Found ("manifest for thepicklenat0r/armv7-hassio-ngrok:2.1.5 not found: manifest unknown: manifest unknown")

**To Reproduce**
Steps to reproduce the behavior:
1. Go to 'AddOn Store'
2. Click on 'Ngrok Client'
3. Scroll down to '.Install'
4. See error



**System**

* Supervisor version: supervisor-2021.01.4 
* Host system version:  core-2021.1.4

**Comment by minskmail** [comment]
Same error

Supervisor version: supervisor-2021.01.4
Host system version: core-2021.1.4

**Comment by sunyalux** [comment]
Me too, Same error!

Failed to install addon
404 Client Error for http+docker://localhost/v1.40/images/create?tag=2.1.5&fromImage=thepicklenat0r%2Farmv7-hassio-ngrok: Not Found ("manifest for thepicklenat0r/armv7-hassio-ngrok:2.1.5 not found: manifest unknown: manifest unknown")

**Comment by Papsnoek** [comment]
Hi Guys,

Maybe just grabbing at straws, but not something related to the recent Security issue HA has addressed?

https://www.home-assistant.io/blog/2021/01/14/security-bulletin/

I wish I could assist to fix, but unfortunately, my programming skills are not good with HA and Python.

**Comment by noamalffasy** [comment]
I don't think that this is it.
This update is specifically about custom integrations, not custom add-ons.
[It was also said in the comments](https://community.home-assistant.io/t/security-bulletin/268456/75)

**Comment by noamalffasy** [comment]
Okay the issue isn't with the add-on, it's with supervisor.

Joining the beta channel and updating to 2021.01.5 fixed the issue.

**Comment by noamalffasy** [comment]
Nevermind it comes back up when I try to start it

**Comment by noamalffasy** [comment]
Another issue pops up later:

```
404 Client Error for http+docker://localhost/v1.40/containers/create?name=addon_1ded40ef_ngrok: Not Found ("No such image: thepicklenat0r/amd64-hassio-ngrok:2.1.5")
```

**Comment by Papsnoek** [comment]
@noamalffasy, I thought the same! LOL.

I update HA to 2021.1.4 and now the Add-On "Installs", but if I inspect Docker in Portainer, it seems the **Docker Image** is getting downloaded, but the **Docker Container** is not getting Created. So definitely something the HA Devs are changing in the latest updates around this is having an affect on this Add-On.

**Comment by Papsnoek** [comment]
@ThePicklenat0r , I have figured out the Issue :) But unfortunately, I think you will need to fix it.

It seems that the "v" in the version number is causing the issues. The HA Devs must have added some code to strip out the "v" in the Version for the Docker image.

So I.e. when I install the Add-On, it creates the Docker Image with "thepicklenat0r/armv7-hassio-ngrok:**v**2.1.5", however, when you start the Add-On, it tries to Use the Image named "thepicklenat0r/armv7-hassio-ngrok:2.1.5". (No "v" in the Docker Image Version).

I think you might need to remove the "v" from you docker Images, or see if this was a desired affect from the HA Devs, or a Bug?

I manage to Confirm this by Fork'ing your Git Repository, and changing the Version to "latest", and now it all works again, but it using the "latest" Docker Image, instead of the "v2.1.5" Docker Image.

**Comment by ustunsoz** [comment]
Still not able to install via add-on store

**Comment by noamalffasy** [comment]
@Papsnoek That's also what I found, it's also what 2021.01.5 tried to fix (which is why it's possible to install the add-on with it). It seems like the `v` is stripped away at some point but I can't quite figure out *when*.

Regardless, I believe that it's an issue that should be solved in hypervisor and not the add-on as other add-ons could have [the same issue](https://github.com/home-assistant/supervisor/issues/2442).

**Comment by Papsnoek** [comment]
@noamalffasy , Nice find!! :) So Yeah, seems a problem with HA Supervisor then.

So guess we will need to wait on a fix from them, or Alternatively, @ThePicklenat0r can make an amendment to his Docker Images to only use Numbers in the Version number?

**Comment by ThePicklenat0r** [comment]
Good find! I agree that this should be solved in HA Supervisor, but I'll release a new version shortly that gets rid of the `v`. It doesn't really serve a purpose and isn't semver compliant anyways. Not sure why I put it there in the first place lol

**Comment by ThePicklenat0r** [comment]
Just released 2.1.6 and I was able to install it successfully. Can anyone else confirm that resolved it for them?

**Comment by Papsnoek** [comment]
Just install and all seems to be up and running with no issues :)

Thanks so much @ThePicklenat0r !! This is by far the single most useful Add-on I have installed as my ISP is blocking Port Forwarding, so this Add-on is a "Must" for external access.

Really appreciate your time 👍

**Comment by ustunsoz** [comment]
After Supervisor update I am able to Install add-on.

However when I start Add-on getting below error and not starting!!!
**Failed to start addon**
404 Client Error for http+docker://localhost/v1.40/containers/create?name=addon_1ded40ef_ngrok: Not Found ("No such image: thepicklenat0r/armv7-hassio-ngrok:2.1.5")

**Comment by ThePicklenat0r** [comment]
@Papsnoek Great to hear! I understand that pain, my old ISP was the reason I made this. Thankfully I was recently able to move to a new ISP that allows port forwarding.

@ustunsoz it looks like you are still using version 2.1.5. In the add-on store make sure you're installing 2.1.6. If it still has the old version you can click the three dots and "Reload" from within the store page and it should find the new version.

**Comment by ustunsoz** [comment]
I have reload add-ons library and the new version came.
Thanks a lot...

**Comment by ThePicklenat0r** [comment]
Glad it is working now and thank you @Papsnoek and @noamalffasy for narrowing down the cause

**Comment by bigmat73** [comment]
All is OK After update Thank you so much !!Le 18 janv. 2021 16:36, Dylan Hasler <notifications@github.com> a écrit :
Closed #23.

**Comment by Papsnoek** [comment]
Hi @ustunsoz ,

It seems you HA is still trying to install version 2.1.5 instead of version 2.1.6.
Try Removing the Add-On Completely (copy the config details to a notepad if you have them set already), then on the "Supervisor" page, go to the Tab "Add-On Store" and on the Top Right, click the menu (3 dots) and select "Reload". Go to the Add-On and install as per normal.

After that, I believe HA will then install version 2.1.6 which is the working version for now.
